server/service/campaigns.go

```diff
@@ -211,6 +211,16 @@ func (svc *Service) NewDistributedQueryCampaignByIdentifiers(ctx context.Context
 		return nil, ctxerr.Wrap(ctx, err, "finding label IDs")
 	}
 
+	// DetectMissingLabels will return the list of labels that are not found in the database
+	// These labels are considered invalid
+	invalidLabels := fleet.DetectMissingLabels(labelMap, labels)
+	if len(invalidLabels) > 0 {
+		setAuthCheckedOnPreAuthErr(ctx)
```

Comment:
This is required because 
`server/service/middleware/authzcheck/authzcheck.go#AuthzCheck`  https://github.com/fleetdm/fleet/blob/main/server/service/middleware/authzcheck/authzcheck.go#L55 returns a 500 forbidden instead of the real error.

```
server/service/middleware/authzcheck/authzcheck.go:53
    testing_client.go:272: 
        	Error Trace:	/Users/ksykulev/projects/fleet/server/service/testing_client.go:272
        	            				/Users/ksykulev/projects/fleet/server/service/testing_client.go:279
        	            				/Users/ksykulev/projects/fleet/server/service/testing_client.go:224
        	            				/Users/ksykulev/projects/fleet/server/service/testing_client.go:289
        	            				/Users/ksykulev/projects/fleet/server/service/integration_live_queries_test.go:1037
        	Error:      	Not equal: 
        	            	expected: 400
        	            	actual  : 500
        	Test:       	TestIntegrationsLiveQueriesTestSuite/TestCreateDistributedQueryCampaign
        	Messages:   	Fleet jsonError: {Message:forbidden Code:0 Errors:[map[name:base reason:forbidden]] UUID:0441fdd8-9cbd-4d94-a484-fb8eb13316dc}
```

Reply:
Deferring to @lucasmrod on proper behavior here. Feels like we should maybe pull auth earlier in this process so we aren't leaking which hosts/labels are valid.

Reply:
That would be ideal.
Currently auth validation happens at the end of the method in `NewDistributedQueryCampaign`
https://github.com/fleetdm/fleet/blob/main/server/service/campaigns.go#L220
on line 43 in the call to `StatusLiveQuery`
https://github.com/fleetdm/fleet/blob/main/server/service/campaigns.go#L43

Reply:
That said, it doesn't hurt at all and is good security practice to add:
```
if err := svc.authz.Authorize(ctx, &fleet.Label{}, fleet.ActionRead); err != nil {
	return nil, err
}
```
before executing `svc.ds.LabelIDsByName`.